### PR TITLE
Add support for hi-res images to AbstractImage cellrenderer

### PR DESCRIPTION
**Pull request by hkollmann**
Based on https://github.com/qooxdoo/qooxdoo/pull/9610. Moved findHigh… ResolutionSource and _getHighResolutionSource to qx.util.ResourceManager.

### CODE REVIEW

File: framework/source/class/qx/ui/basic/Image.js [hunk 1]
```diff
@@ -514,7 +514,7 @@ qx.Class.define("qx.ui.basic.Image",
         }
       }
       else if (ResourceManager.has(source)) {
-        var highResolutionSource = this._findHighResolutionSource(source);
+        var highResolutionSource = qx.util.ResourceManager.getInstance().findHighResolutionSource(source);
```
Comment (johnspackman) [inline framework/source/class/qx/ui/basic/Image.js]: `qx.util.ResourceManager.getInstance()` has already been saved as `ResourceManager`, maybe shorten it here

File: framework/source/class/qx/ui/mobile/basic/Image.js [hunk 2]
```diff
@@ -148,8 +145,9 @@ qx.Class.define("qx.ui.mobile.basic.Image",
         var uri = resourceManager.toUri(source);
 
         if (resourceManager.has(source)) {
-          var highResSource = this._findHighResolutionSource(source);
+          var highResSource = qx.util.ResourceManager.getInstance().findHighResolutionSource(source, qx.ui.mobile.basic.Image.ROOT.getAppScale());
```
Comment (johnspackman) [inline framework/source/class/qx/ui/mobile/basic/Image.js]: also `resourceManager` here

**Review by johnspackman — CHANGES_REQUESTED** [review]

**Review by level420 — APPROVED** [review]

**Review by oetiker — APPROVED** [review]

**Review by cboulanger — APPROVED** [review]

**Review by johnspackman — APPROVED** [review]

**Review by oetiker — APPROVED** [review]

**Review by derrell — APPROVED** [review]

**Comment by level420** [comment]
plenty of days passed. 5 approvals. merging.
thank you @hkollmann

File: framework/source/class/qx/ui/table/cellrenderer/AbstractImage.js [hunk 3]
```diff
@@ -233,21 +233,33 @@ qx.Class.define("qx.ui.table.cellrenderer.AbstractImage",
...
       // set image
       if (this.__imageData.url) {
         content = qx.bom.element.Decoration.create(
-          this.__imageData.url,
+          srcUrl,
           this.getRepeat(),
           {
           width: this.__imageData.width + "px",
           height: this.__imageData.height + "px",
+          "background-size": backgroundSize,
```
Comment (oetiker) [inline framework/source/class/qx/ui/table/cellrenderer/AbstractImage.js]: that seems odd ... decorations would normally use camel case

File: framework/source/class/qx/ui/table/cellrenderer/AbstractImage.js [hunk 4]
```diff
@@ -233,21 +233,33 @@ qx.Class.define("qx.ui.table.cellrenderer.AbstractImage",
     {
       var content = "<div></div>";
 
+      // background size is critical for high-resolution images
+      var backgroundSize = this.__imageData.width + "px, " + this.__imageData.height + "px";
```
Comment (oetiker) [inline framework/source/class/qx/ui/table/cellrenderer/AbstractImage.js]: background-size has no comma: https://developer.mozilla.org/en-US/docs/Web/CSS/background-size #9646

**Pull request merged**